### ISSUE TITLE
Explore skeletonization filters (or similar approaches) to extract the centerline of nerve rootlets segmentation

### ISSUE THREAD

**Issue by jcohenadad** [ivadomed/model-spinal-rootlets#59]

**Comment by maxradx** [comment]
This issue is about exploration of skeletonization filters (or similar approaches) to extract the centerline of nerve rootlets segmentation.

An assumption is that nerve rootlets are tubular structures on medical images, so filtering approaches should work on tubular structures.

1- Multiple filtering modules exist in 3D Slicer. Examples from Slicer documentation[ here](https://slicer.readthedocs.io/en/latest/user_guide/modules/index.html#filtering).

2- It is possible to extract skeleton of a binary object (link [here](https://slicer.readthedocs.io/en/latest/user_guide/modules/extractskeleton.html#skeleton-skeleton-parameters)). However, there does not seem to have default 3D configurations. Would it work on 3D structures? 

3- An extension from which the centerline can be extracted from tubular structure is VMTK (Vascular Modeling Toolkit). Initially designed for vascular structures, it is claimed to work on any tubular structures (like vessels, nerve rootlets ...) A potential user should consider that the last commit on the main branch of github has been 2 months ago (from June 24th, 2024). Github repository [here](https://github.com/vmtk/SlicerExtension-VMTK). Moreoever, when the keyword 'centerline' is used in the 3D Slicer extension manager, only SlicerVMTK is in the results.
<img width="1487" alt="image" src="https://github.com/ivadomed/model-spinal-rootlets/assets/142258732/32a19042-f32a-49f4-b853-3335ea72a007">

4- Retrospectively, I asked chatGPT 4-o the question about exploring skeletonization filters to extract the centerline of nerve rootlets segmentation. Although there might have some errors, I found its answer useful for providing a way of how performing centerline extraction from nerve rootlets in 3D Slicer (joint text file). 

[20240624_chatgpt_answer.txt](https://github.com/user-attachments/files/15961450/20240624_chatgpt_answer.txt)

**Comment by jcohenadad** [comment]
The [tubular extension](https://discourse.slicer.org/t/segment-editor-draw-tube/6050) from Slicer3D seems to work quite well, and is being used for labeling lumbar rootlets. 

We might want to consider re-thinking ground truth labeling method for cervical rootlets as well, but that would imply re-doing all GTs, which is a big task...

**Comment by valosekj** [comment]
I agree that the [tubular extension](https://discourse.slicer.org/t/segment-editor-draw-tube/6050) is great for lumbar rootlets, especially when creating them from scratch.

For the cervical rootlets, the situation is slightly different: in most cases, we get a relatively good initial prediction and only need to correct a few slices, sometimes only a few pixels. This can be done relatively quickly even with FSLeyes/ITKsnap.
But agree, we should keep the tubular extension in mind even for cervical rootlets. Let's see how many corrections we need for MP2RAGE images (https://github.com/ivadomed/model-spinal-rootlets/issues/63#issuecomment-2218257267).
